.github/workflows/npm-publish.yml

```
@@ -0,0 +1,74 @@
+name: NPM Publish Release Workflow
+
+on:
+  push:
+    paths:
+      - "app/client/packages/design-system/ads/**"
+    branches:
+      - "release"
+    
+jobs:
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+    steps:
+      # Get yarn cache directory and set it in a variable
+      - name: Get yarn cache path
+        id: yarn-cache-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+      
+      # Checkout branch or head commit
+      - name: Checkout branch
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      
+      # Set up node version
+      - name: Setup Node js version
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16.x
+      
+      # Load yarn cache
+      - name: Load Yarn cache
+        uses: actions/cache@v3
+        with:
+          path: ${{ steps.yarn-cache-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+      
+      # Install the dependencies
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
```

```suggestion
        run: yarn install --immutable
```

It seems you used something old as an example. Please check how we do the same thing in other CI files.